recipe/build.sh

```diff
@@ -3,12 +3,17 @@
 set -exuo pipefail
 
 mkdir -p build-target
+
+if [[ ${cuda_compiler_version} != "None" ]]; then
+    export CMAKE_ARGS="-DUSE_CUDA=ON ${CMAKE_ARGS}"
```

Comment:
I see that you have mentioned nccl in the patch, you might want to add `USE_NCCL=ON`, and optionally `BUILD_WITH_SHARED_NCCL=ON` to avoid the static library.